src/event_form/components/calendar/base-calendar.controller.js

```diff
@@ -57,6 +57,10 @@ function BaseCalendarController(calendar, $scope) {
     if (calendarType === 'single' && _.isEmpty(calendar.timeSpans)) {
       initTimeSpans();
     }
+
+    if (calendarType === 'single' || calendarType === 'multiple') {
+      timeSpanChanged();
```

Comment:
https://github.com/cultuurnet/udb3-angular/blob/4d66d4dd470fb2c837202699aa1b0099f861dbf3/src/event_form/components/calendar/base-calendar.controller.js#L127-L130

`timeSpanChanged` also calls `setType()` so this might cause a recursive loop? That could be the reason the tests are failing.